js/client/modules/@arangodb/arango-collection.js

```diff
@@ -128,17 +128,34 @@ let appendSyncParameter = function (url, waitForSync) {
   return url;
 };
 
+let appendOverwriteModeParameter = function (url, mode) {
+  if (mode) {
+    if (url.indexOf('?') === -1) {
+      url += '?';
+    }else {
+      url += '&';
+    }
+    url += 'overwriteMode=' + mode;
+  }
+  return url;
+};
+
+
 // //////////////////////////////////////////////////////////////////////////////
 // / @brief append some boolean parameter to a URL
 // //////////////////////////////////////////////////////////////////////////////
 
-let appendBoolParameter = function (url, name, val) {
+let appendBoolParameter = function (url, name, val, when_undefined) {
```

Comment:
this entire `when_undefined` business does not seem necessary, because the only caller that makes use of it makes sure that the value is never undefined... So this code complication can be safely reverted.